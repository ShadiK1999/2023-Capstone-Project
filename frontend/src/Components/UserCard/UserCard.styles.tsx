import { style } from 'typestyle';

export const CardContainer = style({
  backgroundColor: '#bfccb5',
  borderRadius: '6px',
  padding: '20px',
  width: 'fit-content',
  height: 'fit-content',
  display: 'grid',
  gridTemplateColumns: '1fr 1fr',
});

export const UserInfoContainer = style({
  display: 'flex',
  flexDirection: 'column',
  justifyContent: 'center',
  alignItems: 'left',
  gap: '5px',
  paddingLeft: '30px',
});

export const UserName = style({
  fontSize: '1.2rem',
});

export const UserInfo = style({
  fontSize: '0.85rem',
});
