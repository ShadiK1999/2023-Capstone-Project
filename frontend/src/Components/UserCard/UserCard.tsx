import { ProfilePicture } from '..';
import { FC } from 'react';
import * as styles from './UserCard.styles';

type ComponentProps = {
  name: string;
  email: string | null;
  phone: string;
};

const UserCard: FC<ComponentProps> = ({ name, email, phone }) => {
  return (
    <>
      <div className={styles.CardContainer}>
        <div>
          <ProfilePicture name={name} size="Large" />
        </div>
        <div className={styles.UserInfoContainer}>
          <div className={styles.UserName}>{name}</div>
          <div className={styles.UserInfo}>{email}</div>
          <div className={styles.UserInfo}>{phone}</div>
        </div>
      </div>
    </>
  );
};

export { UserCard };
