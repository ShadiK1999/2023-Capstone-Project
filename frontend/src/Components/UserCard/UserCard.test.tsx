// eslint-disable-next-line import/named
import { render } from '@testing-library/react';
import { UserCard } from './UserCard';
import { describe, it, expect } from 'vitest';

describe('User Cards', () => {
  it('matches DOM Snapshot', () => {
    const tree = render(
      <UserCard name={'John Doe'} email={'jdoe@email.com'} phone={'0412345678'} />,
    );
    expect(tree).toMatchSnapshot();
  });

  it('Holds user info ', () => {
    const tree = render(
      <UserCard name={'John Doe'} email={'jdoe@email.com'} phone={'0412345678'} />,
    );
    expect(tree.getByText('John Doe')).toBeInTheDocument();
  });
});
