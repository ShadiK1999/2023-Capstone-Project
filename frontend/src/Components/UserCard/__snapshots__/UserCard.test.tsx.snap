// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`User Cards > matches DOM Snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="fga25bv"
      >
        <div>
          <div
            class="f1lazpvh"
          >
            J
          </div>
        </div>
        <div
          class="fvqs59d"
        >
          <div
            class="f12igrzn"
          >
            John Doe
          </div>
          <div
            class="f3xul3h"
          >
            jdoe@email.com
          </div>
          <div
            class="f3xul3h"
          >
            0412345678
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="fga25bv"
    >
      <div>
        <div
          class="f1lazpvh"
        >
          J
        </div>
      </div>
      <div
        class="fvqs59d"
      >
        <div
          class="f12igrzn"
        >
          John Doe
        </div>
        <div
          class="f3xul3h"
        >
          jdoe@email.com
        </div>
        <div
          class="f3xul3h"
        >
          0412345678
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
